src/useLocalStorage.ts

```diff
@@ -47,6 +53,29 @@ const useLocalStorage = <T>(
     }
   });
 
+  if (options?.sync) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+      const callback = (e: StorageEvent) => {
+        if (e.key !== key) {
+          return;
+        }
+        if (e.newValue === null) {
+          return remove();
+        }
+        try {
+          setState(deserializer(e.newValue));
```

Comment:
After having tried subscribing to storage events in a local variant of this hook, I've found this to be super unstable. It works fine for primitives (boolean, number, string), but not for arrays and objects which will cause an infinite update loop.

A subscribed/synced feature of this hook really needs to be tested thoroughly to make sure it's stable. Would probably need to do some sort of deep-equals check or something to prevent infinite updates.